specs/tree-view/tree-view.md

```diff
@@ -0,0 +1,195 @@
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/tree-view)
+- [Material UI](https://material-ui.com/components/tree-view/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/trees/)
+- [Ant Design](https://ant.design/components/tree/)
+- [Atlassian](https://atlaskit.atlassian.com/packages/core/tree/example/static-tree)
+- [Windows (UWP)](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/tree-view)
+
+---
+
+## Design
+
+*Describe the design of the component, thinking through several perspectives:*
+
+- *A customer using the component on a web page.*
+- *A developer building an app with the component and interacting through HTML/CSS/JavaScript.*
+- *A designer customizing the component.*
+
+### API
+
+*The key elements of the component's public API surface:*
+
+**Tree View**
+
+- extends HTML Element props
+
+**Tree View Item**
+
+- expanded - nodes with child-nodes can either be expanded or collapsed. 
+- selected - the node is selected
+- content - the content for the tree view
+- before-content - slot which precedes content, often a glyph or icon
+- after-content - slot which follows content, often a glyph or icon
+- expand-collapse-glyph - slot for svg, will include a default svg
+- on-expanded-change - callback for when the expanded state has changed
+- on-selected - callback for when an item has been selected
+
+### Anatomy and Appearance
+
+*Screenshots and/or description of the basic appearance of the component. Outline its structure with a diagram of its visual tree (shadow dom). Enumerate key areas of visual customization, such as:*
+
+**Tree View**
+
+![](./images/tree-view.png)
+
+The tree view has no named slots or parts - it has a default slot for children.
+
+Flat tree:
+```html
+<fast-tree-view>
+    <fast-tree-item>Root node one</fast-tree-item>
+    <fast-tree-item>Root node two</fast-tree-item>
+    <fast-tree-item>Root node three</fast-tree-item>
+    <fast-tree-item>Root node four</fast-tree-item>
+</fast-tree-view>
+```
+
+Nested tree:
+```html
+<fast-tree-view>
+    <fast-tree-item>Root node one</fast-tree-item>
+    <fast-tree-item>Root node two</fast-tree-item>
+    <fast-tree-item>
+        <fast-tree-item>Nested node one</fast-tree-item>
+        <fast-tree-item>Nested node two</fast-tree-item>
+    </fast-tree-item>
+    <fast-tree-item>Root node three</fast-tree-item>
+</fast-tree-view>
+```
+
+**Tree view item**
+
+Default
+![](./images/tree-item.png)
+
+Selected
+![](./images/tree-item-selected.png)
+
+Expanded
+![](./images/tree-item-expanded.png)
+
+Collapsed
+![](./images/tree-item-collapsed.png)
+
+Like tree view, the tree view item has a default slot for its children.
+
+Slots:
+- `content`
+- `before-content`
+- `after-content` 
+- `expand-collapse-glyph`
+
+Parts:
+- root
+- content
+- before-content
+- after-content
+- expand-collapse-glyph
+- child-nodes
+- expanded
+- selected
```

Comment:
Good call on `expanded` and `selected` as parts. I'm not sure I would have gone that route but it makes sense when you think about targeting styles at the HTML parts that are expanded/selected.

Reply:
need that custom state API you were talking about @EisenbergEffect!

Reply:
Also this might be a little odd because it is the element itself that is expanded, collapsed, selected, or unselected, not a "part" of the element

```css
fast-tree-view-item::part(selected) {
  /* This actually targets some DOM node *inside* the tree-view-item, not the item itself
}
```

if we had custom states, I'd expect to be able to write:
```css
tree-view-item:selected { /* target item when selected */ }
tree-view-item:selected::part(content) { /* target a part in item when selected */ }
```